src/Ploi/Resources/Server.php

```diff
@@ -103,9 +103,61 @@ public function create(
         return $response->getData();
     }
 
-    public function createCustom(): stdClass
+    public function createCustom(string $ip, array $options): stdClass
     {
         $endpoint = $this->getEndpoint() . '/custom';
+
+        $defaults = [
+            'ip' => $ip,
+            'type' => 'server',
+            'database_type' => 'mysql',
+            'php_version' => '7.4',
+        ];
+
+        $options = [
+            'body' => json_encode(array_merge($defaults, $options)),
+        ];
+
+        $response = $this->getPloi()->makeAPICall($endpoint, 'post', $options);
+
+        $data = $response->getJson();
+        $this->setId($data->id);
+
+        return $data;
+    }
+
+    public function startInstallation(string $url = null)
+    {
+        $id = $this->getId();
+
+        if (!$url && !$id) {
+            throw new RequiresId("This endpoint requires an ID. Supply an ID or a valid installation url.");
+        }
+
+        $endpoint = $url ?: $this->getEndpoint() . "/custom/{$id}/start";
+
+        $response = $this->getPloi()->makeAPICall($endpoint, 'post');
```

Comment:
I feel like this way of setting then immediately getting the endpoint is unnecessary for a couple reasons.

For one, I don't really see the point of setting a property to a value only to immediately get that same value from the object's property when a local variable would cause less confusion and lead to less bugs.

Second, f you're not careful because of inheritance and keeping the same object instance, values can get weird and wrong if you're not careful.

Say we're doing this:

```php
$server = $ploi->servers($serverId);
$keys = $server->sshKeys();
$versions = $server->phpVersions();
```

If `sshKeys` and `phpVersions` sets the endpoint using this:

```php
$this->setEndpoint("{$this->endpoint}/{$this->getId()}/other-path");
```

... it works. But only because the `Server` instance's `$endpoint` property never changes since the `setEndpoint` is inherited. However, if they get set with:

```php
$this->setEndpoint("{$this->getEndpoint()}/{$this->getId()}/other-path");
```

The paths get merged because of inheritance. To me this could all be avoided just setting a local variable.

```php
// This seems awkward
$this->setEndpoint("{$this->endpoint}/{$this->getId()}/other-path");
$this->getPloi()->makeAPICall($this->getEndpoint());

// These seem more reasonable and avoid any potential inheritance pitfalls
$endpoint = "{$this->endpoint}/{$this->getId()}/other-path"
$this->getPloi()->makeAPICall($endpoint);

// or just this...
$this->getPloi()->makeAPICall("{$this->endpoint}/{$this->getId()}/other-path");
```

All that to say, I think it's an unnecessary risk doing `setEndpoint()` just to call `getEndpoint()` immediately after. My 2 cents.